### DISCUSSION TITLE
not working at Kali 2022.4

### DISCUSSION THREAD

**Discussion by Tograse** [pablosnt/rekono#135] [Help]
After the nmap step, all utilities are in the requested status and there is no work. On which version can the software be tested stably?

**Comment by pablosnt** [comment]
Hi @Tograse, it should work in Kali 2022.4, can you send me more information please? Did the nmap execution create findings? How do you install and deploy Rekono? The requested status means that the executions don't start, so it could be a problem with the `executions-queue`.

**Reply by Tograse** [reply]
Good afternoon! I installed it via docker, nmap showed the ports, but after that all utilities are in the execution waiting status

**Reply by Tograse** [reply]
I attach a link to the photo of the error https://dropmefiles.com/uSGOK

**Reply by pablosnt** [reply]
Thank you! Yes, you are right, it's strange. Can you check if there are any errors in the `docker-compose` logs? Specialy in `executions-worker` and `tasks-worker` containers. Anyway, if you are using Docker, is not a problem in Kali Linux environments, because Rekono and the tools are executed within Docker.

**Reply by Tograse** [reply]
**"executions-worker"**
2023-02-09 07:21:22,219 [ERROR] - 16 worker -  -  - [Job 13064690-77da-4130-93f6-59e081feef85]: exception raised while executing (executions.queue.consumer.consumer)
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/rq/worker.py", line 1111, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.10/dist-packages/rq/job.py", line 923, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.10/dist-packages/rq/job.py", line 946, in _execute
    result = self.func(*self.args, **self.kwargs)
  File "/code/executions/queue/consumer.py", line 53, in consumer
    tool_runner.run(targets=targets, previous_findings=previous_findings)       # Tool execution
  File "/code/tools/tools/base_tool.py", line 510, in run
    self.parse_output_file()                                            # Parse output file
  File "/code/tools/tools/nmap.py", line 340, in parse_output_file
    self.parse_nse_scripts(h.scripts_results, technology if len(h.services) == 1 else None, technologies)
  File "/code/tools/tools/nmap.py", line 272, in parse_nse_scripts
    parser(script, tech)                                            # Process NSE result
  File "/code/tools/tools/nmap.py", line 179, in parse_smb_shares
    anonymous = fields.get("Anonymous access")
AttributeError: 'str' object has no attribute 'get'

task-worker - no error

**Reply by pablosnt** [reply]
It's an error during the parsing of nmap output. I was trying to reproduce the error, but I didn't get it. I guess that it is a very specific error that depends on your nmap results. Please, can you send me your nmap report to rekono.project@gmail.com? Please anonymize your output before send it to me.

To get the nmap output, you can check the internal file where it's saved in the `executions-worker` logs. Please, send me these file. In the next picture you have an example:
<img width="1120" alt="Captura de pantalla 2023-02-09 a las 11 54 13" src="https://user-images.githubusercontent.com/69458381/217794215-fcfd04ab-8c43-4fbe-a7ce-e29f01154722.png">

**Reply by Tograse** [reply]
I can't find the specified path /rekono/reports...

**Reply by pablosnt** [reply]
It should be within the `executions-worker` Docker container

**Reply by Tograse** [reply]
executions-worker does not contain the report file specified in the screenshot
![image](https://user-images.githubusercontent.com/112467851/217809256-2a170d01-c204-4393-b924-f8d2c8ca355a.png)

**Reply by pablosnt** [reply]
The screenshot that I have sent is just an example. The filenames are generated randomly. So you need to check the exact filepath in the executions worker logs, where  the executed command appears and it includes the exact filepath where the output is stored (like in the screenshot)

**Reply by pablosnt** [reply]
Hi @Tograse, can I help you with this issue?

**Comment by pablosnt** [comment]
I'm closing this discussion, please, reopen it if you need it. Thanks!
